### PR TITLE
Update the system gems before doing anything else

### PR DESCRIPTION
**Pull request by jyaworski**
@bastelfreak

### CODE REVIEW

**Comment by rnelson0** [comment]
I dare say, we should solve these issues in A module before pushing it to modulesync configs. We should also link to an issue or provide a description of the what and why behind the changes.

**Comment by jyaworski** [comment]
@rnelson0 while toying with #126 @bastelfreak and myself discovered that the version of bundle Travis uses is ancient (1.7.6). Almost all of us develop on the latest version (1.11.2). The same could be said with rubygems. That's what this PR is fixing.

**Comment by bastelfreak** [comment]
As a side note, only old ruby versions ship bundle 1.7.6, ruby2.2 and 2.3 don't have this issue.

File: moduleroot/.travis.yml [hunk 1]
```diff
@@ -21,7 +21,12 @@ addons:
     <%- end -%>
   <%- end -%>
 <%- end -%>
-before_install: rm Gemfile.lock || true
+before_install:
+  - bundle -v
+  - rm Gemfile.lock || true
+  - gem update --system
```
Comment (bastelfreak) [inline moduleroot/.travis.yml]: do we really want to update everything or just bundler? This could bring some sideeffects.

Comment (bastelfreak) [inline moduleroot/.travis.yml]: this is recommended by travis. I'm fine with this.

**Comment by bastelfreak** [comment]
For reference, I tested this successful multiple times in https://github.com/voxpupuli/puppet-collectd/pull/476

**Comment by rnelson0** [comment]
What is broken that is being fixed? What is the performance impact this has?

Being a little overly pedantic here but would like to ensure that we are tracking why we do things here, and that anyone using modulesync on their own can review the voxpupuli configs and understand if they need the same changes in their own configs.

**Comment by bastelfreak** [comment]
sure sure :)

@jyaworski could you update the commit message with some infos? We've to install certain version of gems on certain ruby platforms. This is detailed explained in https://github.com/voxpupuli/modulesync_config/pull/125 . There are several ways to declare the the needed ruby + gem version in a Gemfile. the nice ones require a bundler version that isn't present in every default env from travis.

**Comment by bastelfreak** [comment]
We also need to update bundler. Can you add the line like [here](https://github.com/voxpupuli/puppet-collectd/pull/476/commits/d5b1b18df0cf1deece4b8785f6222d6fca9d2394)?

**Comment by jyaworski** [comment]
@bastelfreak done.

**Comment by bastelfreak** [comment]
I'm fine with this. @rnelson0 are you too?

**Comment by rnelson0** [comment]
What is the performance impact this has? Is this necessary for all modules or just when you need 'listen' to be 3.0.6, and if the latter, when is that?

**Comment by jyaworski** [comment]
@rnelson0 this is brought on by the 'listen' thing, but not confined to it. This brings only performance benefits and brings our testing tooling in-line with our dev tooling. We all test on the latest rubygems/bundle, so Travis should run that.

**Comment by bastelfreak** [comment]
@jyaworski same opinion here. I only checked a single run, gem update took 4.7s, bundler update 4.1. Gem install time went down from 14s to 12s.

**Comment by jyaworski** [comment]
@bastelfreak @rnelson0 I believe this can be merged given the above metric.

**Pull request merged**